raft/server.cc

```diff
@@ -907,6 +898,10 @@ future<> server_impl::io_fiber(index_t last_stable) {
                     std::exchange(_stepdown_promise, std::nullopt)->set_value();
                 }
                 if (!_current_rpc_config.contains(server_address{_id})) {
+                    if (_conf_commit_promise) {
+                        std::exchange(_conf_commit_promise, std::nullopt)->set_exception(commit_status_unknown());
```

Comment:
Why do you need to set commit status unknown here?

Reply:
@gusev-p

Reply:
by analogy with drop_waiters, if we lost leadership for some other reason, not related to a configuration change, we need to release the pending set_configuration

Reply:
I mean, we can loose leadership at any moment, because of connection problems or something, by [this code ](https://github.com/scylladb/scylla/blob/master/raft/fsm.cc#L495) for example

Reply:
OK